api/http/server.go

```diff
@@ -897,6 +916,83 @@ func (s *Server) HandleGetFile(w http.ResponseWriter, r *http.Request) {
 	http.ServeContent(w, r, fileName, time.Now(), newBufferedReadSeeker(reader, getFileBufferSize))
 }
 
+// HandlePin takes a root hash as argument and pins a given file or collection in the local Swarm DB
+func (s *Server) HandlePin(w http.ResponseWriter, r *http.Request) {
+	postPinCount.Inc(1)
+	ruid := GetRUID(r.Context())
+	uri := GetURI(r.Context())
+	fileAddr := uri.Address()
+	log.Debug("handle.post.pin", "ruid", ruid, "uri", r.RequestURI)
+
+	if fileAddr == nil {
+		postPinFail.Inc(1)
+		respondError(w, r, "missig hash to pin ", http.StatusBadRequest)
+		return
+	}
+
+	isRaw := true
```

Comment:
This is more a user experience comment. If isRaw is by default true and setting it to false under specific conditions, there may be some cases that user will provide a value that may suggest false, but interpreted as true, such as "False", "FALSE", "off", "no". I think that a reverse check would be better, assume false, and set to true only if IsRaw has value "true", and even maybe rename the get parameter from `IsRaw` to `is-raw` or just `raw`. It is also possible to support a shorthand `bzz-pin:?raw` (without any parameter) by checking the query map key directly. Just a suggestion.

Reply:
ok. I should have made isRaw as false by default because when uploading in Swarm we upload it as a collection by default. When not uploading it as a collection, then the user has to supply isRaw as true. Also if nothing is provided we should consider isRaw as false. So in summary.. isRaw is true only if it is present and equal to "true" or "True" or any variant of true.

Reply:
Thanks.